docs/internal/input-validations-framework-readme.md

```diff
@@ -0,0 +1,92 @@
+# How to use the framework
+
+The constraints framework implementation is located in the utils/constraints-utils.ts module
+
+```typescript
+import * from 'utils/constraints-utils';
+```
+
+## What can you use with the framework
+
+You will find in the constraints framework the generic calls for the following classes if you properly import them from the constraints framework.
+
+## Validations file constraints-utils.ts
+
+This class holds the supported constraints and function(s) to validate constraints defined below in the rest of this document.
+
+## StringConstraint
+
+Constructor: 
+```typescript
+new StringConstraint(minValue, maxValue);
+```
+
+API reference ['here'](/docs/api/classes/utils.StringConstraint.html)
+
+If given string length falls outside of these inclusive bounds throws detailed Zod error
+
+## UrlStringConstraint
+
+Constructor: 
+```typescript
+new UrlStringConstraint(minValue, maxValue);
+```
+
+API reference ['here'](/docs/api/classes/utils.UrlStringConstraint.html)
+
+If given string length falls outside of these inclusive bounds, or does not follow a proper URL format it throws detailed Zod error
+
+## NumberConstraint
+
+Constructor: 
+```typescript
+new NumberConstraint(minValue, maxValue);
+```
+
+API reference ['here'](/docs/api/classes/utils.NumberConstraint.html)
+
+If given number falls outside of these inclusive bounds throws detailed Zod error.
+
+## ArrayConstraint
+
+Constructor:
+```typescript
+new utils.ArrayConstraint(minValue, maxValue);
+```
+
+API reference ['here'](/docs/api/classes/utils.ArrayConstraint.html)
+
+If given array length falls outside of these inclusive bounds throws detailed Zod error.
+
+## validateConstraints Function
+
+```typescript
+validateConstraints<T>(constraints: ConstraintsType<T>, context: string, ...object: any)
+```
+This is the entry point to use the framework. This function can validate either a single object or an array of objects against the provided constraints.
+
+## How to use the constraints-utils.ts
+
+You need two things when utilizing constraints-utils.ts and the following examples are from ['here'](/lib/stacks/eks-blueprint-stack.ts)
+
+First you need a class with specified keys assigned to given constraints. 
+
+Example with two keys: 
+
+```typescript
+export class BlueprintPropsConstraints implements ConstraintsType<EksBlueprintProps> {
+    id = new StringConstraint(1, 63);
+    name = new StringConstraint(1, 63);
+```
+
+Second you need to call validateConstraints.
```

Comment:
validateConstraints is code so should be in ``.

Reply:
Like this?

Second you need to call 
```typescript 
validateConstraints();
```

Example:

```typescript
validateConstraints(new BlueprintPropsConstraints, EksBlueprintProps.name, blueprintProps);
```

Reply:
Second, you need to call `validateConstraints` function: 
(note: punctuation, formatting)